app/src/main/java/org/mozilla/focus/utils/AndroidBundleSerializer.java

```diff
@@ -0,0 +1,151 @@
+package org.mozilla.focus.utils;
+
+import android.os.Bundle;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class AndroidBundleSerializer {
+
+    private final Map<String, DataTypeHandler> dataTypeHandlers;
```

Comment:
I am not pretty sure, can we just use 'Class' as key for this map?

Reply:
Use String as key can reduce traversal effort to check the exact type of the `SerializedItem`. If we use `Class` object as key and serialize to file, it would need to use `instanceof` to see if it's a byte[] or String class.